Sanitycheck: don't remove "test_" from test case name.

LeiW000 opened the pull request:
Add "test_" back to test case name so that test case name is same
as "c" function name. It allows us to easily find code of a test
case with test case name shown with sanitycheck.

For example:

- kernel.fifo.usage.test_single_fifo_play

Signed-off-by: Steven Wang <steven.l.wang@linux.intel.com>

PerMac commented:
You will also need to modify the failing tests of Sanitycheck TestSuite in this PR, to be in sync with the new naming. 

I see no objections. This change will probably require modifying external CIs (I mean e.g. CI I am taking care of internally) which are based on sanitycheck results, to update their parsers. IMO this change can increase the traceability of tests but I don't have a strong opinion on this. Now my parser has to remove "test_" from detected tests to match the results and it doesn't seem to be a big deal.

nashif reviewed (changes requested):
not sure why you are doing this. This is an identifier and has nothing to do with the test function name.

Commit:
Sanitycheck: don't remove "test_" from test case name.

Add "test_" back to test case name so that test case name is same
as "c" function name. It allows us to easily find code of a test
case with test case name shown with sanitycheck.

For example:

- kernel.fifo.usage.test_single_fifo_play

Signed-off-by: Steven Wang <steven.l.wang@linux.intel.com>

LeiW000 commented:
> You will also need to modify the failing tests of Sanitycheck TestSuite in this PR, to be in sync with the new naming.
> 
> I see no objections. This change will probably require modifying external CIs (I mean e.g. CI I am taking care of internally) which are based on sanitycheck results, to update their parsers. IMO this change can increase the traceability of tests but I don't have a strong opinion on this. Now my parser has to remove "test_" from detected tests to match the results and it doesn't seem to be a big deal.

@PerMac 
Yes, I will address that issue if there is no strong objection on my proposal. Thanks for the comments, which is helpful to me. I don't know much of background of sanitcheck design. I'm thinking if it is correct thing to do.

LeiW000 commented:
> not sure why you are doing this. This is an identifier and has nothing to do with the test function name.

@nashif 
Because, I would like to make it easy to find result of test case with the test case name in the doc, which is generated with doxygen.

Take "test_single_fifo_play()" for example, the function name is shown in the doxygen doc. When someone want to find its result in testing report generated with sanitycheck, they would just search it with "test_single_fifo_play" ranther than "single_fifo_play". So I'm thinking if we should add "test_" back in sanitycheck testing result.

nashif commented:
> Because, I would like to make it easy to find result of test case with the test case name in the doc, which is generated with doxygen.

you can do this programmatically. 
The identifier is being used also for things that are not covered by functions and adding "test_" to each identifier is really redundant and noisy.

LeiW000 commented:
Decide to cancel this PR after discussion with Anas.
